src/main/java/org/jabref/logic/ai/chatting/model/Gpt4AllModel.java

```diff
@@ -0,0 +1,144 @@
+    private final HttpClient httpClient;
+    private final Gson gson = new Gson();
+
+    public Gpt4AllModel(AiPreferences aiPreferences, HttpClient httpClient) {
+        this.aiPreferences = aiPreferences;
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> list) {
+        LOGGER.debug("Generating response from Gpt4All model with {} messages: {}", list.size(), list);
+
+        List<Message> messages = list.stream()
+                .map(chatMessage -> switch (chatMessage) {
+                    case AiMessage aiMessage -> new Message("assistant", aiMessage.text());
+                    case SystemMessage systemMessage -> new Message("system", systemMessage.text());
+                    case ToolExecutionResultMessage toolExecutionResultMessage -> new Message("tool", toolExecutionResultMessage.text());
+                    case UserMessage userMessage -> new Message("user", userMessage.singleText());
+                    default -> throw new IllegalStateException("Unknown ChatMessage type: " + chatMessage);
+                }).collect(Collectors.toList());
+
+        TextGenerationRequest request = TextGenerationRequest
+                .builder()
+                .model(aiPreferences.getSelectedChatModel())
+                .messages(messages)
+                .temperature(aiPreferences.getTemperature())
+                .max_tokens(2048)
+                .build();
+
+        try {
+            String requestBody = gson.toJson(request);
+            String baseUrl = aiPreferences.getSelectedApiBaseUrl();
+            String fullUrl = baseUrl.endsWith("/") ? baseUrl + "chat/completions" : baseUrl + "/chat/completions";
+            HttpRequest httpRequest = HttpRequest.newBuilder()
+                    .uri(URI.create(fullUrl))
+                    .header("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(requestBody))
+                    .timeout(Duration.ofSeconds(60))
+                    .build();
+
+            HttpResponse<String> response = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+            LOGGER.info("Gpt4All response: {}", response.body());
+
+            TextGenerationResponse textGenerationResponse = gson.fromJson(response.body(), TextGenerationResponse.class);
+            if (textGenerationResponse.choices() == null || textGenerationResponse.choices().isEmpty()) {
+                throw new IllegalArgumentException("No choices returned in the response");
+            }
+
+            String generatedText = textGenerationResponse.choices().getFirst().message().content();
+            if (generatedText == null || generatedText.isEmpty()) {
+                throw new IllegalArgumentException("Generated text is null or empty");
+            }
+            return new Response<>(new AiMessage(generatedText), new TokenUsage(0, 0), FinishReason.OTHER);
```

Comment:
Add a comment there, saying that we never check the token usage and finish reason and our class is not a 100% correct implementation of langchain4j's `ChatLanguageModel`, and we only implemented the functionality we need.

Just in case someone will wonder why this is 0 :)

Reply:
Ok, I will fix them later. Thanks for reviewing my code.